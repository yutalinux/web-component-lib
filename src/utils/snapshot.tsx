import { type Attributes, type FunctionComponent, createElement } from "react";
import { renderToString } from "react-dom/server";

export function snapshotElement<T extends object>(
  type: FunctionComponent<T>,
  args: (Attributes & T) | null,
) {
  const element = createElement(type, args);
  return renderToString(element);
}
