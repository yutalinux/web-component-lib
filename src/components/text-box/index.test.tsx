import { expect, test } from "bun:test";
import { snapshot } from "@/utils/tests/snapshot";
import TextBox from ".";

test("TextBox Snapshot", () => {
  expect(
    snapshot(TextBox, {
      placeholder: "Placeholder",
    }),
  ).toMatchSnapshot();
  expect(
    snapshot(TextBox, {
      type: "password",
    }),
  ).toMatchSnapshot();
});
