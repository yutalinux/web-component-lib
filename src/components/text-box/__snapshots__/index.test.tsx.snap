// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox Snapshot 1`] = `"<input class="bg-white text-black focus:bg-neutral-100 dark:bg-neutral-900 dark:text-white dark:border-neutral-800 dark:focus:bg-neutral-800 border box-border rounded-md px-2 py-1 w-full" placeholder="Placeholder" data-rac=""/>"`;

exports[`TextBox Snapshot 2`] = `"<input type="password" class="bg-white text-black focus:bg-neutral-100 dark:bg-neutral-900 dark:text-white dark:border-neutral-800 dark:focus:bg-neutral-800 border box-border rounded-md px-2 py-1 w-full" data-rac=""/>"`;
